spec/services/task_service/gpt_generate_tests_spec.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe TaskService::GptGenerateTests do
+  describe '.new' do
+    subject(:gpt_generate_tests_service) { described_class.new(task:) }
+
+    let(:programming_language) { build(:programming_language, language: 'Ruby') }
+    let(:task) { build(:task, description: 'Sample Task', programming_language:) }
+
+    it 'assigns task' do
+      expect(gpt_generate_tests_service.instance_variable_get(:@task)).to be task
+    end
+
+    it 'assigns message from task description' do
+      expect(gpt_generate_tests_service.instance_variable_get(:@message)).to eq('Sample Task')
+    end
+
+    it 'assigns language from programming language' do
+      expect(gpt_generate_tests_service.instance_variable_get(:@language)).to eq('Ruby')
+    end
+  end
+
+  describe '#execute' do
+    let(:programming_language) { create(:programming_language, language: 'Python') }
+    let(:task) { create(:task, description: 'Create a Python script.', programming_language:) }
+    let(:gpt_generate_tests_service) { described_class.new(task:) }
+    let(:mock_client) { instance_double(OpenAI::Client) }
+
+    before do
+      allow(OpenAI::Client).to receive(:new).and_return(mock_client)
+      allow(mock_client).to receive(:chat).and_return('choices' => [{'message' => {'content' => 'def test_script(): assert True'}}])
+    end
+
+    it 'creates a test file and test instance related to the task' do
+      expect { gpt_generate_tests_service.execute }.to change(TaskFile, :count).by(1).and change(Test, :count).by(1)
+
+      test_file = TaskFile.last
+      test = Test.last
+
+      expect(test_file.content).to eq('def test_script(): assert True')
+      expect(test_file.name).to eq('test.py')
+      expect(test.title).to eq('AI-generated Test')
+      expect(task.tests).to include(test)
+    end
+  end
+
+  describe '#gpt_response' do
```

Comment:
Since this method should be private (see other comment). This example could be part of the perform example.

Reply:
Hey :) I am not sure what you mean here. could you explain more? I have changed the method above to private as you mentioned.

Reply:
I wouldn't test this method directly, but through the normal execution path (`Service.call`) (see other comment)